### PR TITLE
Fix for 509

### PR DESCRIPTION
**Pull request by magnatelee**

### CODE REVIEW

File: legate/core/solver.py [hunk 1]
```diff
@@ -350,21 +350,21 @@ def compute_launch_shape(
         partitions: dict[PartSym, PartitionBase],
         all_outputs: set[Store],
         unbound_ndim: Optional[int],
-    ) -> Optional[Shape]:
+    ) -> Tuple[Optional[Shape], bool]:
```
Comment (rohany) [inline legate/core/solver.py:353]: I would add a comment to this now explaining what the boolean return value is.

File: legate/core/solver.py [hunk 2]
```diff
@@ -476,45 +541,28 @@ def cost(unknown: PartSym) -> tuple[int, bool]:
...
+                constraints,
+                must_be_even,
+            )
 
-            for to_align in cls:
-                if to_align in partitions:
+            launch_shape, done = self.compute_launch_shape(
+                result, all_outputs, unbound_ndim
+            )
+            if can_retry and not done:
```
Comment (rohany) [inline legate/core/solver.py:562]: This is a bit of tricky logic that could use a comment.

**Review by rohany — APPROVED** [review]
Logic looks good to me, just some requests for comments.

File: legate/core/solver.py (hunk 2 above)
Comment (magnatelee) [reply legate/core/solver.py:573]: https://github.com/nv-legate/legate.core/pull/510/commits/595965ec41013f28447c8294b7f8811ff327c763

File: legate/core/solver.py (hunk 1 above)
Comment (magnatelee) [reply legate/core/solver.py:356]: https://github.com/nv-legate/legate.core/pull/510/commits/595965ec41013f28447c8294b7f8811ff327c763

**Review by rohany — APPROVED** [review]

**Pull request merged**